### ISSUE TITLE
Syntax highlighting – improve contrast on light and dark themes

### ISSUE THREAD

**Issue by squidfunk** [squidfunk/mkdocs-material#1862]
## Description

The syntax highlighting doesn't quite work well with the `slate` theme, as some contrasts are too low. Accessibility guidelines suggest a contrast ratio of foreground:background from 4.5:1.

### Use Cases

- Syntax highlighting works for visually impaired people
- Syntax highlighting is readable in low light situations

### Screenshots / Mockups

<img width="704" alt="Bildschirmfoto 2020-08-11 um 09 03 22" src="https://user-images.githubusercontent.com/932156/89867468-8711b380-dbb1-11ea-971a-74978a341c71.png">

**Comment by StarfallProjects** [comment]
Note that the syntax highlighting also doesn't meet AA standards with the light/default theme (pale grey background to code).

**Comment by squidfunk** [comment]
Yes, I know. I'm happy for any help on this topic. Otherwise, I'll tackle it as soon as I find some time.

**Comment by DeNeutoy** [comment]
Hi @squidfunk,

Thanks for porting this issue! 

A couple of concrete examples of what I was finding difficult to read:

1) Inline code examples are hard to distinguish from the rest of the text
![image](https://user-images.githubusercontent.com/16001974/89933860-8c2c2e00-dbc4-11ea-88ad-2a0a73e797f0.png)

2) The RHS menu contrast is tough to see:
![image](https://user-images.githubusercontent.com/16001974/89933952-b5e55500-dbc4-11ea-9a0c-62298bf1016a.png)

3) This contrast on this indented text is quite low (i'm not sure what formatting produces this, sorry)
![image](https://user-images.githubusercontent.com/16001974/89934147-ffce3b00-dbc4-11ea-8b2f-d65805453a79.png)


But overall I think this is more to do with the choice of colours, and I completely get that some people want dark themes etc! I think the manual toggle from #1732  would completely solve my issues with this theme, btw.

Thanks for the excellent library!

**Comment by squidfunk** [comment]
31bbd76c improves syntax highlighting for code blocks to have a contrast ratio of 4.5+:1. Unfortunately, that makes colors a lot less vivid and I can imagine that people will complain about it, but we can give it a try. As an alternative, we might still move this to a separate color scheme. The TOC blurred link color has also been set to a higher contrast.

**Comment by squidfunk** [comment]
@DeNeutoy 4a96f62e improves contrast for inline code blocks, they should now better stand out. Blockquotes already have a contrast ratio of 4.5:1. Furthermore, this is the style for blockquote which we have since version 1 – changing it may upset some people. This is, after all, an opinionated template, but colors can be changed very easily, as the guide details 😊

**Comment by squidfunk** [comment]
Released as part of 5.5.7.
